### ISSUE TITLE
2.3.0 Release

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#799]
Can be labelled **ready** when all other issues on this milestone are closed.

- [x] run code analysis in VS in *Release* mode and address violations (send a regular PR which must be merged before continuing)
- [x] if necessary, change `VERSION_SUFFIX` on the [CI Server](http://teamcity.codebetter.com/admin/editBuildParams.html?id=buildType:bt929)
      to appropriate "-beta123" or "" (for non-betas) value and initiate a build
- [x] check build
- edit draft release in [GitHub UI](https://github.com/FakeItEasy/FakeItEasy/releases):
    - [x] ensure completeness of release notes, including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
    - [x] attach main nupkg and/or analyzer nupkg, whichever have content to release
    - [x] publish the release
- [x] push nupkg(s) to NuGet
- [x] de-list pre-release or superseded buggy NuGet packages if present
- [x] update website with contributors list (if in place)
- [x] tweet, mentioning contributors and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- [x] run `rake next_version[new_version]` to
    - create a pull request that changes the version in CommonAssemblyInfo.cs to the expected version (of form _xx.yy.zz_)
    - create a new draft GitHub Release
    - create a new milestone for the next release
    - create a new issue (like this one) for the next release, adding it to the new milestone
- [ ] close this milestone

**Comment by blairconrad** [comment]
Things are moving along. Do we want to release once we get #851 and #840 in?
Then we could consider 3.0, as we've been holding off some breaking changes.
(Or not consider 3.0. That works too.)

**Comment by thomaslevesque** [comment]
> Things are moving along. Do we want to release once we get #851 and #840 in?

:+1:

**Comment by TimLovellSmith** [comment]
@thomaslevesque @blairconrad  I notice that all other issues in this milestone are closed, would love to see a beta of 2.3 get out. I ran the code analysis in release mode - there were some warnings, prompting me to submit another PR: #869

**Comment by thomaslevesque** [comment]
@TimLovellSmith we'll release 2.3 soon, we just needed to fix the warnings first (thanks for the help!). We don't usually make betas for minor versions, so it will be a stable release.

**Comment by blairconrad** [comment]
Folks, I've checked off the first few steps, now that @TimLovellSmith and @thomaslevesque have cleaned up the warnings. I'm sleepy, so won't continue now.
We like to publicize releases on weekdays, but I'm happy to continue the steps on the weekend or on Monday as pleases everyone.

**Comment by thomaslevesque** [comment]
Thanks @blairconrad!

Yes, we should probably wait until Monday for the rest of the steps

**Comment by blairconrad** [comment]
Release tweet:

https://twitter.com/fakeiteasyfx/status/780355336427474944

**Comment by blairconrad** [comment]
Release:
https://github.com/FakeItEasy/FakeItEasy/releases/tag/2.3.0

**Comment by blairconrad** [comment]
Created [release issue for 2.4.0](https://github.com/FakeItEasy/FakeItEasy/issues/873), assuming we don't jump to 3.0.
